dvc/output.py

```diff
@@ -419,21 +446,29 @@ def cache_path(self):
         )
 
     def get_hash(self):
+        _, hash_info = self._get_hash_meta()
+        return hash_info
+
+    def _get_hash_meta(self):
         if self.use_cache:
             odb = self.odb
             name = self.odb.fs.PARAM_CHECKSUM
         else:
             odb = self.repo.odb.local
             name = self.fs.PARAM_CHECKSUM
-        _, _, obj = build(
+        _, meta, obj = build(
             odb,
-            self.fs_path,
+            self._fs_path,
```

Comment:
We use `_fs_path` here since for `get_hash()`/`get_meta()` we want the checksum and metadata for the current version of the output